python/lsst/drp/tasks/metadetection_shear.py

```diff
@@ -0,0 +1,286 @@
+                        "unit": "",
+                    },
+                ),
+                pa.field(
+                    "patch_y",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Row within the tract of the patch on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                pa.field(
+                    "cell_x",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Column within the patch of the cell on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                pa.field(
+                    "cell_y",
+                    pa.uint64(),
+                    nullable=False,
+                    metadata={
+                        "doc": "Row within the patch of the cell on which this measurement was made.",
+                        "unit": "",
+                    },
+                ),
+                # TODO: add more field definitions here
+            ]
+        )
+
+    def runQuantum(
+        self,
+        qc: QuantumContext,
+        inputRefs: InputQuantizedConnection,
+        outputRefs: OutputQuantizedConnection,
+    ) -> None:
+        # Docstring inherited.
+        # Read the coadds and put them in the order defined by
+        # config.required_bands (note that each MultipleCellCoadd object also
+        # knows its own band, if that's needed).
+        coadds_by_band = {
+            ref.dataId["band"]: qc.get(ref) for ref in inputRefs.input_coadds
+        }
+        outputs = self.run([coadds_by_band[b] for b in self.config.required_bands])
+        qc.put(outputs, outputRefs)
+
+    def run(self, patch_coadds: Sequence[MultipleCellCoadd]) -> Struct:
+        """Run metadetection on a patch.
+
+        Parameters
+        ----------
+        patch_coadds : `~collections.abc.Sequence` [ \
+                `~lsst.cell_coadds.MultipleCellCoadd` ]
+            Per-band, per-patch coadds, in the order specified by
+            `MetadetectionShearConfig.required_bands`.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            Structure with the following attributes:
+
+            - ``object_catalog`` [ `pyarrow.Table` ]: the output object
+              catalog for the patch, with schema equal to `object_schema`.
+        """
+        single_cell_tables: list[pa.Table] = []
+        for single_cell_coadds in zip(*patch_coadds, strict=True):
+            single_cell_tables.append(self.process_cell(single_cell_coadds))
+        # TODO: if we need to do any cell-overlap-region deduplication here
+        # (instead of purely in science analysis code), this is where'd it'd
+        # happen.
+        return Struct(
```

Comment:
Does this struct have to match the schema returned by  make_object_schema?

Reply:
The `object_catalog` attribute of the struct needs to be a pyarrow table that matches that schema.  The attributes of the `Struct` itself correspond to the `cT.Output` connections in `MetadetectionShearConnections`, of which there is only one right now.